README.md
## Welcome to GitHub Pages

You can use the [editor on GitHub](https://github.com/crackwallsports/testwebsite/edit/master/README.md) to maintain and preview the content for your website in Markdown files.

Whenever you commit to this repository, GitHub Pages will run [Jekyll](https://jekyllrb.com/) to rebuild the pages in your site, from the content in your Markdown files.

### Markdown

Markdown is a lightweight and easy-to-use syntax for styling your writing. It includes conventions for

```markdown
Syntax highlighted code block

# Header 1
## Header 2
### Header 3

- Bulleted
- List

1. Numbered
2. List

**Bold** and _Italic_ and `Code` text

[Link](url) and ![Image](src)
```

For more details see [GitHub Flavored Markdown](https://guides.github.com/features/mastering-markdown/).

### Jekyll Themes

Your Pages site will use the layout and styles from the Jekyll theme you have selected in your [repository settings](https://github.com/crackwallsports/testwebsite/settings). The name of this theme is saved in the Jekyll `_config.yml` configuration file.

### Support or Contact

Having trouble with Pages? Check out our [documentation](https://help.github.com/categories/github-pages-basics/) or [contact support](https://github.com/contact) and we’ll help you sort it out.


articles/News.org
#+TITLE: News
#+AUTHOR: Xt3


* Use
#+BEGIN_SRC lisp
(progn
  ;; Load
  (load "./articles/News")

  ;; Generate
  (->file #P"./articles/News.html" (news-index))
  nil)

;; Date
(format nil "更新时间: ~a" (human-date (get-universal-time)))


#+END_SRC

* Package
#+BEGIN_SRC lisp :tangle yes
(in-package :twb)
#+END_SRC
* Helper
#+BEGIN_SRC lisp :tangle yes
;; (defvar *news-topics* (make-hash-table :test 'equal))
;; (defun news-set-topic (topic value)
;;   (setf (gethash topic *news-topics*) value))
;; (defun news-get-topic (topic)
;;   (gethash topic *news-topics*))


(defparameter *news-topics* nil)
(defun news-add-topic (topic)
  (push topic *news-topics*))

(let ((show t))
  (defun news-topic (title time &rest content)
    (let ((id (format nil "collapse~a" (random 10000))))
      `(div (:class "topic card")
            (div (:class "card-header d-flex w-100 justify-content-between")
                 (button (:class ,(concat "btn btn-link" (if show
                                                             ""
                                                             " collapsed"))
                                 :type "button"
                                 :data-toggle "collapse"
                                 :data-target ,(concat "#" id))
                         (h2 () ,title))
                 (small () ,(format nil "更新时间: ~a" time)))
            (div (:class ,(concat "collapse"
                                  (if show
                                      (progn (setf show nil)
                                             " show")
                                      ""))
                         :id ,id
                         :data-parent "#topic-accordion")
                 (div (:class "card-body")
                      ,@content))))))


;; (defun news-to-topic (title time &rest content)
;;   (news-set-topic title (->html (apply #'news-topic title time content))))

(defun news-to-topic (title time &rest content)
  (news-add-topic (->html (apply #'news-topic title time content))))

(defun @ (link &optional title)
    (let ((tle (if title title link))) 
        `(a (:href ,link) ,tle)))

(defun u (&rest rest)
  `(ul () ,@rest))

(defun :- (&rest rest)
  `(li () ,@rest))




#+END_SRC
* Htm
#+BEGIN_SRC lisp :tangle yes
(defun news-index ()
  (->html
   (layout-template
    nil
    :title "News 新鲜事"
    :links `((link (:rel "stylesheet" :href "/testwebsite/css/bootstrap.min.css"))
             (link (:rel "stylesheet" :href "/testwebsite/css/font-awesome.min.css"))
             (link (:rel "stylesheet" :href "/testwebsite/css/style.css")))
    :head-rest `((style () "
.btn-link {color: black }
.btn-link:hover {text-decoration:none}
cite {
font-size: 88% }
q {
border-left: 5px rgb(210, 212, 212) solid;
display: block;
padding: 5px 10px 5px 10px;
text-align: justify;
}
q::before, q::before {
display: block;
content: \"\";
}
li pre {
display: inline;
margin: 0;
white-space: pre-wrap;
}
li q {
margin-left: 16px;
}

.zoom {      
-webkit-transition: all 0.35s ease-in-out;    
-moz-transition: all 0.35s ease-in-out;    
transition: all 0.35s ease-in-out;     
cursor: -webkit-zoom-in;      
cursor: -moz-zoom-in;      
cursor: zoom-in;  
}     
.zoom:hover,  
.zoom:active,   
.zoom:focus {
-ms-transform: scale(7);    
-moz-transform: scale(7);  
-webkit-transform: scale(7);  
-o-transform: scale(7);  
transform: scale(7);    
position:relative;      
z-index:100;  
}
"))
    :content
    `(,(site-header)
       (main (:class "content")
             ;; 
             (div (:class "accordion" :id "topic-accordion"
                          :style "font-size: 140%")
                  ;; ,@(loop for v being the hash-values of *news-topics*
                  ;;      collect v)
                  ,@(nreverse *news-topics*)))
       ,(site-footer))
    :scripts `((script (:src "/testwebsite/js/jquery-3.2.1.min.js"))
               (script (:src "https://stackpath.bootstrapcdn.com/bootstrap/4.1.3/js/bootstrap.min.js"
                             :integrity "sha384-ChfqqxuZUCnJSK3+MXmPNIyE6ZbWh2IMqE241rYiqJxyMiZ6OW/JmZQ5stwEULTy"
                             :crossorigin "anonymous"))))))
#+END_SRC
* Topics
** 郭文贵
- 信息源
  - 郭媒体 : [[https://www.guo.media/milesguo][@milesguo]]
  - LiveStream : [[https://livestream.com/accounts/27235681/events/8197481][郭文贵中文直播]]
  - Youtube : [[https://www.youtube.com/channel/UCO3pO3ykAUybrjv3RBbXEHw/featured][郭文贵]]
  - Instagram : [[https://www.instagram.com/guowengui/][guowengui]]
- 郭七条
  - 反对 以黑治国 以警治国 以贪反贪 以黑反贪
  - 不反国家 不反民族 不反习主席
  - 修订(增加): 反对以假治国
- 海航王健事件
- 主要内容跟踪
*** HEAD
#+BEGIN_SRC lisp :tangle yes
(defparameter *news-topics-guo* nil)  
#+END_SRC

*** 主要内容
- 2019
**** COMMENT Template
- 1
  #+BEGIN_SRC lisp :tangle yes
(push
 (u (:- '(small () "2019..")
        (u (:- "视频: " (@ "" "")
               '(small () "")
               (u (:- '(pre () ""))))
           (:- "郭文: " (@ "")
               '(q () (pre () ""))
               (u (:- '(pre () "")))))))
 ,*news-topics-guo*)
#+END_SRC

**** 9
#+BEGIN_SRC lisp :tangle yes
(push
 (u (:- (@ "/testwebsite/articles/2019/9/guo-news-201909.html" "2019.9 (-5 7-9 23-) ")))
 ,*news-topics-guo*)
#+END_SRC

**** 8
#+BEGIN_SRC lisp :tangle yes
(push
 (u (:- (@ "/testwebsite/articles/2019/8/guo-news-201908.html" "-2019.8.29")))
 ,*news-topics-guo*)
#+END_SRC

**** 7
#+BEGIN_SRC lisp :tangle yes
(push
 (u (:- (@ "/testwebsite/articles/2019/7/guo-news-201907.html" "2019.7.14-")))
 ,*news-topics-guo*)
#+END_SRC

**** 6
#+BEGIN_SRC lisp :tangle yes
(push
 (u (:- (@ "/testwebsite/articles/2019/6/guo-news-201906.html" "2019.6.-16")))
 ,*news-topics-guo*)
#+END_SRC



**** 5
#+BEGIN_SRC lisp :tangle yes
(push
 (u (:- (@ "/testwebsite/articles/2019/5/guo-news-201905.html" "2019.5")))
 ,*news-topics-guo*)
#+END_SRC

**** 4
#+BEGIN_SRC lisp :tangle yes
(push
 (u (:- (@ "/testwebsite/articles/2019/4/guo-news-201904.html" "2019.4")))
 ,*news-topics-guo*)
#+END_SRC

**** 3
#+BEGIN_SRC lisp :tangle yes
(push

TestWebSite.lisp
(in-package :cl-user)
(defpackage test-website
  (:use :cl 
        :xt3.web.base)
  (:nicknames :twb)
  (:export :generate
           :gen-with-frame))
(in-package :test-website)

(defun generate ()
  (->file #P"./index.html" (index ()))
  (->file #P"./css/style.css" (css))
  (->file #P"./js/main.js" (js)))

(defun gen-with-frame (title path)
  (->file (merge-pathnames (concat "box-" (pathname-name path) ".html")
                           path)
          (->html
           (layout-template
            nil
            :title title
            :links `((link (:rel "stylesheet" :href "/testwebsite/css/bootstrap.min.css"))
                     (link (:rel "stylesheet" :href "/testwebsite/css/font-awesome.min.css"))
                     (link (:rel "stylesheet" :href "/testwebsite/css/style.css")))
            :head-rest `("
<script>
function resizeIframe(obj) {
  obj.style.height = obj.contentWindow.document.body.scrollHeight + 'px';
}
</script>
")
            :content
            `(,(site-header)
               (main (:class "content")
                     (iframe (:src ,(merge-pathnames path
                                                     #P"/testwebsite/")
                                   :onload "resizeIframe(this)"
                                   ;; :scrolling "no"
                                   :style "width:100% ; border:none; box-shadow: 0px 0px 20px; border-radius: 20px;")))
               ,(site-footer))
            :scripts `((script (:src "/testwebsite/js/jquery-3.2.1.min.js"))
                       (script (:src "https://stackpath.bootstrapcdn.com/bootstrap/4.1.3/js/bootstrap.min.js"
                                     :integrity "sha384-ChfqqxuZUCnJSK3+MXmPNIyE6ZbWh2IMqE241rYiqJxyMiZ6OW/JmZQ5stwEULTy"
                                     :crossorigin "anonymous")))))))

(defun human-date (date)
  (and date
       (multiple-value-bind
             (second minute hour day month year)
           (decode-universal-time date)
         (format nil "~4D.~2,'0D.~2,'0D ~2,'0D:~2,'0D:~2,'0D"
                 year month day hour minute second))))

(defun layout-template (args &key (title "标题") links head-rest content scripts)
  (declare (ignore args))
  `(,(doctype)
     (html (:lang "en")
           (head ()
                 (meta (:charset "utf-8"))
                 (meta (:name "viewport"
                              :content "width=device-width, initial-scale=1, shrink-to-fit=no"))
                 (meta (:name "description" :content "?"))
                 (meta (:name "author" :content "Xt3"))
                 (title nil ,title)
                 ,@links
                 ,@head-rest)
           (body ()
                 ,@content
                 ,@scripts))))

(defun site-header ()
  '(header (:class "side-header")
    (a (:class "logo" :href "/testwebsite/index.html")
     (img (:src "/testwebsite/resource/carrot.PNG")))
    (nav (:class "contact" :role "navigation")
     (ul ()
      ;; (li () (a (:href "https://www.guo.media/xt3") "GUO.MEDIA"))
      (li () (a (:href "https://twitter.com/iamnotXt3") "Twitter"))
      (li () (a (:href "mailto:crackwallsports@gmail.com") "Email"))))))

(defun site-footer ()
  '(footer (:class "side-footer")
    (a (:class "twitter" :href "https://twitter.com/iamnotXt3")
     (i (:class "fa fa-twitter" :aria-hidden "true") "@iamnotXt3")) (br)
    (a (:class "github" :href "https://github.com/crackwallsports")
     (i (:class "fa fa-github" :aria-hidden "true") "crackwallsports")) (br)
    (a (:class "email" :href "mailto:crackwallsports@gmail.com")
     (i (:class "fa fa-envelope" :aria-hidden "true") "crackwallsports@gmail.com")) (br)
    (p (:class "copyright") "All content copyright © 2017 Xt3")))

(defun index (args)
  (->html
    (layout-template
     args
     :title (or (getf args :title) "Xt3 Blog")
     :links `((link (:rel "stylesheet" :href "/testwebsite/css/bootstrap.min.css"))
              (link (:rel "stylesheet" :href "/testwebsite/css/font-awesome.min.css"))
              (link (:rel "stylesheet" :href "/testwebsite/css/style.css")))
     :head-rest `((style () "
.btn-link {color: black }
.btn-link:hover {text-decoration:none}
cite {
font-size: 88% }
q {
border-left: 5px rgb(210, 212, 212) solid;
display: block;
padding: 5px 10px 5px 10px;
text-align: justify;
}
q::before, q::before {
display: block;
content: \"\";
}
li pre {
display: inline;
margin: 0;
white-space: pre-wrap;
}
li q {
margin-left: 16px;
}

.zoom {      
-webkit-transition: all 0.35s ease-in-out;    
-moz-transition: all 0.35s ease-in-out;    
transition: all 0.35s ease-in-out;     
cursor: -webkit-zoom-in;      
cursor: -moz-zoom-in;      
cursor: zoom-in;  
}     
.zoom:hover,  
.zoom:active,   
.zoom:focus {
-ms-transform: scale(7);    
-moz-transform: scale(7);  
-webkit-transform: scale(7);  
-o-transform: scale(7);  
transform: scale(7);    
position:relative;      
z-index:100;  
}
"))
     :content
     `(,(site-header)
        (main (:class "content")
              ;; articles
              ,@(let ((posts (with-open-file (stream #P"posts.lisp")
                               (read stream))))
                  (loop for i in posts
                     collect (destructuring-bind (time ref title excerpt) i
                               `(article (:class "post")
                                         (header (:class "post-header")
                                                 (span (:class "post-meta")
                                                       (time () ,time))
                                                 (h2 (:class "post-title")
                                                     (a (:href ,ref) ,title)))
                                         (section (:class "post-excerpt")
                                                  (p () ,excerpt)))))))
        ,(site-footer)))))

(defun css ()
  (->css
   `((* ( ;; :border "1px dashed red"
         :box-sizing "border-box"
         :padding 0 :margin 0))
     (html (:font-size "62.5%"))
     (body (:color "#3a4145" ;; :font-family "Georgia, STSong"
                   ))
     (a (:text-decoration "none"))
     ("ul, li" (:list-style "none"
                :padding-left "8px" :list-style-position "inside"))
     ;; Header
     ;; (".side-header" (:position "fixed" :margin "20px auto"
     ;;                            :width "200px" :height "200px")
     ;;                 (".logo" (:position "relative" :left "30px"
     ;;                                     :display "block")
     ;;                          (img (:width "120px" :height "120px")))
     ;;                 (".contact" (:font-size "2rem"  :padding-left "20%")
     ;;                             (li ()
     ;;                                 (a (:color "rgba(30, 30, 30, 0.92)")))))
     (".side-header" (:text-align "center" :margin "20px auto")
                     (".logo" ()
                              (img (:width "120px" :height "120px")))
                     (".contact" (:font-size "2rem")
                                 (li ()
                                     (a (:color "rgba(30, 30, 30, 0.92)")))))
     ;; Main-Content
     (".content" ( :padding-left "20%" :padding-right "10%" :padding-top "30px" :min-height "600px")
                 (".post" (:font-size "1.6rem"
                                      :width "80%"
                                      :margin-top "3rem" :margin-left "2rem"
                                      :padding-bottom "3rem"
                                      :border-bottom "#a7abb3 1px solid"
                                      :word-break "break-word"))
                 (".post-meta" (:display "block"
                                         :margin-bottom "1rem"
                                         :font-size "1.5rem"
                                         :color "#9eabb3"))
                 (".post-title a" (:color "black"))
                 (".post-title a:hover" (:color ,(css-color :indigo)))
                 (".post-excerpt p" (:margin "1.6rem 0" :font-size "1.5rem" :line-height "1.5em"))
                 ;; Topic

                 (".topic" ()
                           ("li p" (:padding-left "16px" :margin "auto"))
                           ("li::before" (:content "\"-\"" :padding-right "8px"))))
     ;; Footer
     (".side-footer" (:margin "4rem 0 0 0" :padding "3rem 0"
                              :text-align "center"
                              :color "#bbc7cc" :background-color "#f9f9f9"
                              :border-top "#ebf2f6 1px solid"
                              :font-size "1.6rem" :line-height "1.6em")
                     (".twitter, .email, .github" (:color "black"
                                                          :display "inline-block"))
                     ;; ("a:hover, span:hover" (:font-size "2rem"))
                     ))))

(defun js () "")
